sw/device/lib/dif/dif_pwrmgr.h

```diff
@@ -0,0 +1,499 @@
+ *
+ * @param config Configuration for initializing a power manager.
+ * @param pwrmgr Internal state of the initialized power manager.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_init(dif_pwrmgr_config_t *config,
+                                    dif_pwrmgr_t *pwrmgr);
+
+/**
+ * Enable or disable low power state.
+ *
+ * When enabled, the power manager transitions to low power state on the next
+ * wait for interrupt (WFI) instruction. Since the hardware clears the
+ * corresponding bit automatically, this function must be called before each
+ * transition to low power state.
+ *
+ * @param pwrmgr A power manager.
+ * @param new_state Whether the power manager should transition to low power
+ *                  state on the next WFI instruction.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_enabled_set(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_toggle_t new_state);
+
+/**
+ * Check if low power state is enabled.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_state Whether the power manager will transition to low power
+ *                  state on the next WFI instruction.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_enabled_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_toggle_t *cur_state);
+
+/**
+ * Low power state configuration.
+ *
+ * Specifies whether various components should be enabled in low power state.
+ */
+typedef struct dif_pwrmgr_low_power_config {
+  /**
+   * Whether the core clock should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t core_clock;
+  /**
+   * Whether the input/output (IO) clock should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t io_clock;
+  /**
+   * Whether the main power domain should be enabled in low power state.
+   */
+  dif_pwrmgr_toggle_t main_power_domain;
+} dif_pwrmgr_low_power_config_t;
+
+/**
+ * Configure low power state.
+ *
+ * @param pwrmgr A power manager.
+ * @param new_config New low power state configuration.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_config_set(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_low_power_config_t *new_config);
+
+/**
+ * Get the current low power state configuration.
+ *
+ * @param pwrmgr A power manager.
+ * @param cur_config Current low power state configuration.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pwrmgr_result_t dif_pwrmgr_low_power_config_get(
+    const dif_pwrmgr_t *pwrmgr, dif_pwrmgr_low_power_config_t *cur_config);
+
+/**
+ * Request types.
+ */
+typedef enum dif_pwrmgr_req_type {
+  /**
+   * A wake-up request.
+   */
+  kDifPwrmgrReqTypeWakeUp,
+  /**
+   * A reset request.
+   */
+  kDifPwrmgrReqTypeReset,
+} dif_pwrmgr_req_type_t;
+
+/**
+ * Lock sources for a request type.
```

Comment:
Could you please explain what "source" is, ctrl + f and searching for "source" in https://docs.opentitan.org/hw/ip/pwrmgr/doc/index.html didn't help much.

Reply:
Thanks for pointing this out. I added a comment that hopefully makes things clearer.